sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -103,6 +108,83 @@
 # Helper functions =============================================================
 #
 
+
+def is_cmis_api(api):
+   return type(api) == CmisApi
+
+
+def get_cmis_application_desired(api, host_lane_count, speed):
+    """
+    Get the CMIS application code that matches the specified host side configurations
+
+    Args:
+        api:
+            XcvrApi object
+        host_lane_count:
+            Number of lanes on the host side
+        speed:
+            Integer, the port speed of the host interface
+
+    Returns:
+        Integer, the transceiver-specific application code
+    """
+
+    if speed == 0 or host_lane_count == 0:
+        return None
+
+    appl_code = None
+    app_found = False
+    if is_cmis_api(api):
+        appl_dict = api.get_application_advertisement()
+        for index in appl_dict.keys():
+            app_info = appl_dict[index]
+            if app_info.get('host_lane_count') != host_lane_count:
+                continue
+            if get_interface_speed(app_info.get('host_electrical_interface_id')) != speed:
+                continue
+            appl_code = index
+            app_found = True
+            break
```

Comment:
How about just return here? And you don't need app_found.

Reply:
I refactored this function a little to make it more readable